src/storage/mod.rs

```diff
@@ -886,52 +915,56 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
                     Self::with_tls_engine(|engine| Self::snapshot(engine, snap_ctx)).await?;
                 {
                     let begin_instant = Instant::now_coarse();
+
                     let stage_snap_recv_ts = begin_instant;
                     let mut statistics = Vec::with_capacity(keys.len());
                     let buckets = snapshot.ext().get_buckets();
-                    let perf_statistics = PerfStatisticsInstant::new();
-                    let snap_store = SnapshotStore::new(
-                        snapshot,
-                        start_ts,
-                        ctx.get_isolation_level(),
-                        !ctx.get_not_fill_cache(),
-                        bypass_locks,
-                        access_locks,
-                        false,
-                    );
-                    let mut stats = Statistics::default();
-                    let result = snap_store
-                        .batch_get(&keys, &mut statistics)
-                        .map_err(Error::from)
-                        .map(|v| {
-                            let kv_pairs: Vec<_> = v
-                                .into_iter()
-                                .zip(keys)
-                                .enumerate()
-                                .filter(|&(i, (ref v, ref k))| {
-                                    metrics::tls_collect_read_flow(
-                                        ctx.get_region_id(),
-                                        Some(k.as_encoded()),
-                                        Some(k.as_encoded()),
-                                        &statistics[i],
-                                        buckets.as_ref(),
-                                    );
-                                    stats.add(&statistics[i]);
-                                    !(v.is_ok() && v.as_ref().unwrap().is_none())
-                                })
-                                .map(|(_, (v, k))| match v {
-                                    Ok(Some(x)) => Ok((k.into_raw().unwrap(), x)),
-                                    Err(e) => Err(Error::from(e)),
-                                    _ => unreachable!(),
-                                })
-                                .collect();
-                            KV_COMMAND_KEYREAD_HISTOGRAM_STATIC
-                                .get(CMD)
-                                .observe(kv_pairs.len() as f64);
-                            kv_pairs
-                        });
-
-                    let delta = perf_statistics.delta();
+                    let (result, delta, stats) = {
```

Comment:
a little verbose, how about using drop(_guard), so no need to use brace

Reply:
rust compiler won't let this happen

Reply:
send check failed